tesseract/tesseract_geometry/test/tesseract_geometry_unit.cpp

```diff
@@ -44,6 +45,207 @@ TEST(TesseractGeometryUnit, Instantiation)  // NOLINT
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getX(), 1, 1e-5);
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getY(), 1, 1e-5);
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getZ(), 1, 1e-5);
+}
+
+TEST(TesseractGeometryUnit, Cone)  // NOLINT
+{
+  using T = tesseract_geometry::Cone;
+  auto geom = std::make_shared<T>(1, 1);
+  EXPECT_NEAR(geom->getRadius(), 1, 1e-5);
+  EXPECT_NEAR(geom->getLength(), 1, 1e-5);
+
+  auto geom_clone = geom->clone();
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getRadius(), 1, 1e-5);
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getLength(), 1, 1e-5);
+}
+
+TEST(TesseractGeometryUnit, Cylinder)  // NOLINT
+{
+  using T = tesseract_geometry::Cylinder;
+  auto geom = std::make_shared<T>(1, 1);
+  EXPECT_NEAR(geom->getRadius(), 1, 1e-5);
+  EXPECT_NEAR(geom->getLength(), 1, 1e-5);
+
+  auto geom_clone = geom->clone();
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getRadius(), 1, 1e-5);
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getLength(), 1, 1e-5);
+}
+
+TEST(TesseractGeometryUnit, Capsule)  // NOLINT
+{
+  using T = tesseract_geometry::Capsule;
+  auto geom = std::make_shared<T>(1, 1);
+  EXPECT_NEAR(geom->getRadius(), 1, 1e-5);
+  EXPECT_NEAR(geom->getLength(), 1, 1e-5);
+
+  auto geom_clone = geom->clone();
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getRadius(), 1, 1e-5);
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getLength(), 1, 1e-5);
+}
+
+TEST(TesseractGeometryUnit, Sphere)  // NOLINT
+{
+  using T = tesseract_geometry::Sphere;
+  auto geom = std::make_shared<T>(1);
+  EXPECT_NEAR(geom->getRadius(), 1, 1e-5);
+
+  auto geom_clone = geom->clone();
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getRadius(), 1, 1e-5);
+}
+
+TEST(TesseractGeometryUnit, Plane)  // NOLINT
+{
+  using T = tesseract_geometry::Plane;
+  auto geom = std::make_shared<T>(1, 1, 1, 1);
+  EXPECT_NEAR(geom->getA(), 1, 1e-5);
+  EXPECT_NEAR(geom->getB(), 1, 1e-5);
+  EXPECT_NEAR(geom->getC(), 1, 1e-5);
+  EXPECT_NEAR(geom->getD(), 1, 1e-5);
+
+  auto geom_clone = geom->clone();
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getA(), 1, 1e-5);
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getB(), 1, 1e-5);
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getC(), 1, 1e-5);
+  EXPECT_NEAR(std::static_pointer_cast<T>(geom_clone)->getD(), 1, 1e-5);
+}
+
+TEST(TesseractGeometryUnit, ConvexMesh)  // NOLINT
+{
+  auto vertices = std::make_shared<tesseract_common::VectorVector3d>();
+  vertices->emplace_back(1, 1, 0);
+  vertices->emplace_back(1, -1, 0);
+  vertices->emplace_back(-1, -1, 0);
+  vertices->emplace_back(1, -1, 0);
+
+  auto faces = std::make_shared<Eigen::VectorXi>();
+  faces->resize(5);
+  (*faces)(0) = 4;
+  (*faces)(1) = 0;
+  (*faces)(2) = 1;
+  (*faces)(3) = 2;
+  (*faces)(4) = 3;
+
+  using T = tesseract_geometry::ConvexMesh;
+  auto geom = std::make_shared<T>(vertices, faces);
```

Comment:
Not related to this, but we need doxygen on these constructors

Reply:
I added documentation to mesh, convex mesh and sdf mesh.